Orange/classification/naive_bayes.py

```diff
@@ -47,23 +48,81 @@ def __init__(self, log_cont_prob, class_prob, domain):
         self.class_prob = class_prob
 
     def predict_storage(self, data):
+        if type(data) is Table:  # pylint: disable=unidiomatic-typecheck
+            return self.predict(data.X)
         if isinstance(data, Instance):
-            data = [data]
-        if len(data.domain.attributes) == 0:
+            return Table(np.atleast_2d(data.x))
+
+        if not len(data) or not len(data[0]):
```

Comment:
`data` can be a `Storage` or a numpy array or, in principle, any sequence of iterables over floats. Thence `len(data[0])`. The original test would (almost) always crash; apparently that line was never executed.